File: app/src/main/java/demo/povarnicin/firstapp/SharedPrefWrapper.java
package demo.povarnicin.firstapp;

import android.content.Context;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.appcompat.app.AppCompatActivity;

public class SharedPrefWrapper {

    private static final String USER_LOGGED_IN = "userLoggedIn";
    private static final String USER_ID = "userLoggedId";
    private Context context;
    private SharedPreferences sPref;

    public SharedPrefWrapper(Context context) {
        this.context = context;
        sPref = context.getSharedPreferences("auth", AppCompatActivity.MODE_PRIVATE);
    }

    void saveUserLoggedIn() {
        SharedPreferences.Editor editor = sPref.edit();
        editor.putBoolean(USER_LOGGED_IN, true);
        editor.apply();
    }

    void setUserLoggedOut() {
        SharedPreferences.Editor editor = sPref.edit();
        editor.putBoolean(USER_LOGGED_IN, false);
        editor.apply();
    }

    boolean checkIfUserLoggedIn() {
        boolean flag = sPref.getBoolean(USER_LOGGED_IN, false);
        return sPref.getBoolean(USER_LOGGED_IN, false);
    }

    void saveId(int id) {
        SharedPreferences.Editor editor = sPref.edit();
        editor.putInt(USER_ID, id);
        editor.apply();
    }

    public int getId() {
        return sPref.getInt(USER_ID, -1);
    }
}

File: app/src/main/java/demo/povarnicin/firstapp/utils/SharedPrefWrapper.kt
package demo.povarnicin.firstapp.utils

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity

class SharedPrefWrapper(private val context: Context) {
    private val sPref: SharedPreferences
    fun saveUserLoggedIn() {
        val editor = sPref.edit()
        editor.putBoolean(USER_LOGGED_IN, true)
        editor.apply()
    }

    fun setUserLoggedOut() {
        val editor = sPref.edit()
        editor.putBoolean(USER_LOGGED_IN, false)
        editor.apply()
    }

    fun checkIfUserLoggedIn(): Boolean {
        val flag = sPref.getBoolean(USER_LOGGED_IN, false)
        return sPref.getBoolean(USER_LOGGED_IN, false)
    }

    fun saveId(id: Int) {
        val editor = sPref.edit()
        editor.putInt(USER_ID, id)
        editor.apply()
    }

    val id: Int
        get() = sPref.getInt(USER_ID, -1)

    companion object {
        private const val USER_LOGGED_IN = "userLoggedIn"
        private const val USER_ID = "userLoggedId"
    }

    init {
        sPref = context.getSharedPreferences("auth", AppCompatActivity.MODE_PRIVATE)
    }
}